### PR TITLE
Event lifecycle: tracker-io engine + create/update/status-event skills

### PR DESCRIPTION
**Pull request by rparundekar**
## Summary

Adds the **event-lifecycle** layer on top of the AAIF meetup skills — the per-event runbook (`Event Tracker.docx`) was the one part of running an event that nothing automated.

Two pieces:

1. **`lib/aaif_meetups/` — a deterministic, stdlib-only docx engine** (no Drive calls, no third-party deps):
   - `office.py` — read/write `word/document.xml` in a `.docx` zip; navigate/edit tables, rows, cells, paragraph runs.
   - `tracker.py` — event model over a tracker (`list_events`/`read_event`), edits (`set_field`/`set_due_dates`/`add_event`), and the shared date-stamping rule (`new_due = new_event_date + (example_task_due − example_event_date)`; clock-time day-of cells are left alone).

2. **Three skills** that operate on a **local** tracker file; each `SKILL.md` instructs the agent to drive the `gws` CLI directly (locate → download → upload):
   - `aaif-event-status` — read-only digest of overdue / due-soon tasks by owner.
   - `aaif-create-event` — clone the example section, fill details, stamp all phase due-dates; dedup-guard; mode is implicit (chapter ⇒ in-person tracker, series ⇒ online tracker).
   - `aaif-update-event` — change-driven edits ("add speaker", "move date"); recomputes due-dates when the date moves and flags stale downstream assets.

Also in this branch (Drive-cleanup groundwork):
- Retired the per-folder `SKILLS.md.docx` references in `create_chapter`/`create_series` docs/docstrings (the 45 Drive copies were trashed and the template trackers re-pointed at this repo separately).

## Design principle

**Python = well-known, unchanging transforms only** (clone / replace / restamp). **Every Google Drive action stays with the agent** via the `gws` CLI, kept flexible — Python never nests gws calls. `create_chapter.py`/`create_series.py` are unchanged from `main` apart from a docstring fix.

## Testing

- `python -m unittest discover -s lib/aaif_meetups/tests` — 14 tests (office round-trip + primitives, date math, event model, writes, clone-and-append with save→reload corruption check).
- Per-skill core tests (`classify`, `apply_local`, `apply_changes`) on real tracker fixtures.
- End-to-end dry-run validated against live Drive (read-only + local edit).

Spec: `docs/superpowers/specs/2026-06-30-aaif-event-lifecycle-design.md` · Plan: `docs/superpowers/plans/2026-06-30-aaif-event-lifecycle.md`

## Out of scope (follow-ups)

File-aware banner/deck generation (#3), chapter/series health (#4), and the two-plugin ops/organizer split (#5).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by rparundekar** [comment]
## Hero Self-Review

Ran all five `pr-review-toolkit` agents; verified the high-impact findings empirically against both fixtures. The stdlib-only + "no gws in Python; agent drives Drive" boundary holds cleanly across every file. Findings below are real correctness issues the 18 passing tests don't cover.

### Critical (1)
- **[silent-failure / comment] `create_event.py:13-16` + `tracker.py:220` — fields silently dropped on online trackers.** `FIELD_MAP` maps `--platform→"VENUE"` and `--location→"LOCATION / CITY"`, but the **online** tracker's labels are `PLATFORM` / `STREAM / JOIN LINK` (verified) — it has no `VENUE`/`LOCATION` row. `add_event` calls `_set_detail` and **discards its `False` (not-found) return**, so `--platform "Zoom"` on an online series prints "Done." while writing nothing. On the IRL tracker, `--venue` and `--platform` both target `VENUE` (last-wins collision). Fix: map `platform→"PLATFORM"`; make `add_event` collect unmatched labels and error (same contract as `set_field`).

### Important (5)
- **[test / code-reviewer] `office.py:8,16-24` — `save_document` re-serializes via ElementTree and renames every namespaced prefix** (`r:id`→`ns2:id`; 313 `ns*:` prefixes after save, verified). URIs are preserved so the file stays namespace-valid, but it's a fidelity regression and `test_roundtrip` only counts `{w}tbl` by URI, so it can't catch this. Fix: scan the root's `xmlns:` decls and `ET.register_namespace` them before serializing.
- **[silent-failure / code-reviewer] `tracker.py:110` — `_select` matches the first title *containing* the query.** Write path: `update_event "AI Night"` can edit "AI Night · Finance Edition" instead and report success. Read path: `event_status` re-selects each event by title (`event_status.py:55`), so when one title is a substring of another, one event prints twice and the other never. Fix: prefer an exact case-insensitive title match; warn/raise on ambiguous substring. For `event_status`, digest the already-resolved event object instead of re-selecting by title.
- **[code-reviewer] `tracker.py:209-212` — cloned event keeps the example's caption and loses its heading.** The span starts at `detail_idx-1`, which is the literal "Example event — duplicate this whole section per night…" paragraph; the real date/title heading (`detail_idx-2`) is excluded. Every created event is captioned "Example event…" with no heading. Fix: include + rewrite the heading paragraph; drop/replace the caption.
- **[silent-failure] `tracker.py:24` — `parse_event_date` defaults a yearless date to the current year.** `--date "January 15"` (today 2026-06-30) → `2026-01-15`, 5 months in the past, shifting all due-dates backward. Reject yearless/ambiguous dates instead of guessing.
- **[comment] `aaif-update-event/SKILL.md:41-42` — detail-label list is in-person only.** Online uses `PLATFORM` / `STREAM / JOIN LINK`; `--set "VENUE=…"` on an online series raises `LookupError`. Document both label sets.

### Suggestions (4)
- **[silent-failure] `tracker.py:88` — a typo'd `DATE & TIME` makes an event silently undated**, dropping it from `next`/`latest` selection with no warning.
- **[code-reviewer] `office.py:51` — `set_cell_text` updates hyperlink display text but not the `.rels` `r:id` target**, so `--luma <new>` (and cloned events) show the new URL but link to the old one.
- **[type-design] Introduce a `Task` dataclass** (`tracker.py:131`). The 4-field task dict is read with `.get("status")` across a file boundary (`event_status.py`), so a typo'd key silently returns a default rather than erroring.
- **[test] Coverage gaps:** multi-event selection (`next`/`latest`/ambiguous substring), year-boundary date math, `set_field` raising on a missing label, the online fixture (downloaded but never loaded), and a save→reload **document.xml fidelity** assertion (not just table count).

### Strengths
- The "Python edits a local docx, agent drives gws" boundary is clean — zero `subprocess`/`gws` calls in any Python (all 5 agents confirmed).
- `add_event` test is behavioral: count delta, title, status reset, restamp, and save→reload.
- `restamp` correctly preserves clock-time `DAY OF` cells and blanks; `update_event`'s restamp-before-overwrite ordering is explicitly tested.
- `save_document` same-path write verified safe (reads all members into memory before opening the write handle).

---
Self-review by [Claude Code](https://claude.ai/code)

**Comment by rparundekar** [comment]
## Hero Self-Review Improvements

All 10 findings (1 Critical, 5 Important, 4 Suggestions) fixed in `73f2605`. 31 tests pass (was 18); pre-commit clean.

**Critical (1 / 1 fixed):**
- `create_event.py` + `tracker.py:add_event` — fields silently dropped on label mismatch → **fixed**: `add_event` now raises listing unmatched labels; `--platform` maps to `PLATFORM` (not `VENUE`), `--join`→`STREAM / JOIN LINK` added. Verified: `--platform` on a chapter tracker now aborts loudly instead of no-op'ing.

**Important (5 / 5 fixed):**
- `office.save_document` re-serialization renamed all namespace prefixes (`r:id`→`ns2:id`) → **fixed**: register the doc's xmlns prefixes before serializing; verified 0 `ns*:` prefixes and 11 `r:id` preserved after save.
- `_select` first-substring match could edit/print the wrong event → **fixed**: exact-title match first, unique substring next, ambiguous substring raises.
- `add_event` kept the "Example event — duplicate…" caption and lost the heading → **fixed**: heading cloned + rewritten to the new date/title; caption dropped.
- `parse_event_date` defaulted yearless dates to the current year → **fixed**: now raises; a year is required.
- update/create `SKILL.md` listed in-person labels only → **fixed**: both docs now document chapter vs online label sets.

**Suggestions (4 / 4 fixed):**
- `event_status` now flags events whose `DATE & TIME` didn't parse (was silently anchored to today).
- `Task` is now a `@dataclass(frozen=True)` — attribute access, so a typo'd field raises instead of returning a `.get()` default.
- Documented the `set_cell_text` hyperlink-target limitation (display text vs `.rels` target) in code + create-event SKILL.md.
- Added the missing tests: multi-event selection, year-boundary date math, raise-on-missing-label, add_event heading/caption, online fixture coverage, save namespace fidelity, empty-cell, combined `--set`+`--date`.

Commit: `73f2605`

---
Applied by [Claude Code](https://claude.ai/code)

File: skills/aaif-create-event/scripts/create_event.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+              "organizer", "location"):
+        ap.add_argument("--" + f)
+    ap.add_argument("--dry-run", action="store_true")
+    a = ap.parse_args()
+
+    event_date = tracker.parse_event_date(a.date)
+    fields = _fields_from_args(a)
+    root = office.read_document(a.docx)
+    if any(a.title.lower() in e["title"].lower() for e in tracker.list_events(root)):
+        sys.exit("ABORT: an event titled %r already exists in %s." % (a.title, a.docx))
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-create-event/scripts/create_event.py]: Fixed in be27df2 — the dedup guard now uses an exact case-insensitive title match (extracted as `title_exists()`), so e.g. "Eval Night" no longer collides with "Eval Night · Builder Series". Covered by a test.

File: skills/aaif-create-event/scripts/create_event.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    ap.add_argument("--title", required=True)
+    ap.add_argument("--date", required=True, help='e.g. "Wed · August 12, 2026 · 18:00 — late"')
+    for f in ("theme", "venue", "platform", "speakers", "luma", "capacity",
+              "organizer", "location"):
+        ap.add_argument("--" + f)
+    ap.add_argument("--dry-run", action="store_true")
+    a = ap.parse_args()
+
+    event_date = tracker.parse_event_date(a.date)
+    fields = _fields_from_args(a)
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-create-event/scripts/create_event.py:57]: Already addressed in 73f2605 — `--platform` now maps to the `PLATFORM` label (distinct from `VENUE`), and a flag whose label is absent on the tracker now raises rather than silently overwriting. So `--venue`/`--platform` can no longer collide: on a chapter tracker `--platform` aborts (no PLATFORM row); on an online tracker `--venue` aborts.

File: skills/aaif-update-event/scripts/update_event.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def apply_changes(root, event, set_pairs, date_str):
+    """Mutate root in place. set_pairs is a list of 'LABEL=VALUE' strings.
+    If date_str is given, recompute all due-dates from the *original* date first,
+    then write the authoritative new DATE & TIME string. Returns the stale-asset set."""
+    stale = set()
+    for pair in set_pairs:
+        label, _, value = pair.partition("=")
+        tracker.set_field(root, event, label.strip(), value.strip())
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-update-event/scripts/update_event.py:26]: Fixed in be27df2 — `apply_changes` now raises a clear `ValueError("--set must be LABEL=VALUE")` when an arg has no '='. Covered by a test.

File: skills/aaif-create-event/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+2. **Download it:**
+
+   ```
+   gws drive files get --params '{"fileId":"<DOC_ID>","alt":"media"}' --output tracker.docx
+   ```
+
+3. **Add the event (deterministic, local).** Aborts if the title already exists:
+
+   ```
+   python skills/aaif-create-event/scripts/create_event.py tracker.docx \
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-create-event/SKILL.md]: Fixed in be27df2 — now invoked via `python3 ${CLAUDE_SKILL_DIR}/scripts/create_event.py` per CONTRIBUTING.md.

File: skills/aaif-update-event/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+   ```
+   gws drive files get --params '{"fileId":"<DOC_ID>","alt":"media"}' --output tracker.docx
+   ```
+
+2. **Apply the change (deterministic, local):**
+
+   ```
+   # add/replace a speaker
+   python skills/aaif-update-event/scripts/update_event.py tracker.docx "Agentic AI Night" \
+     --set "SPEAKER(S)=Jane Doe (Agent Infra)"
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-update-event/SKILL.md]: Fixed in be27df2 — `${CLAUDE_SKILL_DIR}/scripts/update_event.py`.

File: skills/aaif-update-event/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+   # move the date (recomputes all due-dates from the original date)
+   python skills/aaif-update-event/scripts/update_event.py tracker.docx "Agentic AI Night" \
+     --date "Wed · July 8, 2026 · 17:30 — late"
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-update-event/SKILL.md]: Fixed in be27df2 — `${CLAUDE_SKILL_DIR}/scripts/update_event.py`.

File: skills/aaif-event-status/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+2. **Download it** to a temp path:
+
+   ```
+   gws drive files get --params '{"fileId":"<DOC_ID>","alt":"media"}' --output tracker.docx
+   ```
+
+3. **Run the digest** (read-only, local):
+
+   ```
+   python skills/aaif-event-status/scripts/event_status.py tracker.docx ["event"]
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply skills/aaif-event-status/SKILL.md]: Fixed in be27df2 — `${CLAUDE_SKILL_DIR}/scripts/event_status.py`.

File: lib/aaif_meetups/tracker.py [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+    example = events[0]
+    old_date = example["date"]
+    body = _body(root)
+    kids = list(body)
+    # span: the paragraph immediately before the detail table .. the last phase table
+    detail_idx = kids.index(example["detail_table"])
+    start = detail_idx - 1 if detail_idx > 0 and kids[detail_idx - 1].tag == office.W + "p" else detail_idx
+    last = example["phase_tables"][-1] if example["phase_tables"] else example["detail_table"]
+    end = kids.index(last)
+    block = [copy.deepcopy(kids[i]) for i in range(start, end + 1)]
```
[the remark this answers is not in the document]
Comment (rparundekar) [reply lib/aaif_meetups/tracker.py]: Already addressed in 73f2605 — `add_event` now clones the heading paragraph, rewrites it to the new EVENT TITLE / DATE & TIME, and drops the example "duplicate this whole section" caption.

**Comment by rparundekar** [comment]
## Respond-to-PR Improvements

Addressed all 8 Copilot review comments (6 code/doc fixes; 2 were already fixed by the earlier self-review commit `73f2605`). All threads resolved.

**Code changes (3 applied):**
- `create_event.py:dedup` — substring match wrongly aborted distinct titles → now an **exact** case-insensitive match (`title_exists()`), with a test. (`be27df2`)
- `update_event.py:apply_changes` — `--set` without `=` could silently clear a field → now raises `ValueError("--set must be LABEL=VALUE")`, with a test. (`be27df2`)
- `aaif-{create,update}-event` + `aaif-event-status` `SKILL.md` — scripts now invoked via `${CLAUDE_SKILL_DIR}/scripts/...` per CONTRIBUTING.md so they work when installed outside the repo. (`be27df2`)

**Already fixed in `73f2605` (resolved):**
- `--venue` / `--platform` both → `VENUE`: now `--platform`→`PLATFORM` (distinct), and a flag whose label is absent on the tracker raises — they can no longer silently collide.
- `add_event` cloned heading not updated: now rewrites the heading to the new EVENT TITLE / DATE & TIME and drops the example caption.

Commits: `73f2605`, `be27df2`

Remaining unresolved threads: none

---
Applied by [Claude Code](https://claude.ai/code)
